### PR TITLE
Remove tree artifacts

### PR DESCRIPTION
**Pull request by restingbull**
Small optimization for kotlin compilation. May result in improvements for remote builds.

Depends on: bazelbuild/rules_kotlin/pull/288

### CODE REVIEW

**Comment by restingbull** [comment]
@googlebot I consent.

**Comment by cgruber** [comment]
I consent. (In case it's me)

**Comment by cgruber** [comment]
I consent

**Comment by cgruber** [comment]
@googlebot I consent.

File: kotlin/internal/jvm/plugins.bzl [hunk 1]
```diff
@@ -19,40 +19,51 @@ load(
...
-        "annotation_processors": "a serializeable list of structs containing annotation processor definitions",
+        "annotation_processors": "depset of structs containing annotation processor definitions",
     },
 )
 
-_EMPTY_PLUGIN_INFO = [KtJvmPluginInfo(annotation_processors = [])]
+_EMPTY_PLUGIN_INFO = [KtJvmPluginInfo(annotation_processors = depset())]
 
+# Mapping functions for args.add_all.
+# These preserve the transitive depsets until needed.
```
Comment (cgruber) [inline kotlin/internal/jvm/plugins.bzl]: Wait... they do? I thought `[x for x in y]` was a list comprehension. This doesn't convert `y` into a list?

**Comment by cgruber** [comment]
I'm LGTM on these two subsequent commits, but let's get the other one merged and rebase this on that, so we can make sure I'm looking at the correct set of commits.

File: kotlin/internal/jvm/plugins.bzl (hunk 1 above)
Comment (restingbull) [reply kotlin/internal/jvm/plugins.bzl]: If you use args.add_all(<depset<Foo>>, map_each=fooToString) the depset isn't flattened until the command line is needed. 

However, this looks like a partial change. I think I merged it wrong. They should be processor.class and processor.cp

I'll go look.

Comment (restingbull) [reply kotlin/internal/jvm/plugins.bzl]: A commit snuck on the wrong pr. Resolve.

**Comment by cgruber** [comment]
Ok, so post-hoc LGTM I guess. :)

**Pull request merged**